### PR TITLE
initial implementation of MooD via DEH9000

### PR DESCRIPTION
**Pull request by jmtd**

### CODE REVIEW

**Comment by jmtd** [comment]
Thanks for the comments. I've implemented most of your suggestions.

File: examples/mood.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+from deh9000 import *
+
+# pacifist monsters
+for monster in mobjinfo:
+    if not (monster.flags & MF_COUNTKILL or monster == mobjinfo[MT_SKULL]):
+        continue
+
+    if monster.missilestate:
+        states[monster.missilestate].tics = 1
```
Comment (fragglet) [inline examples/mood.py]: Is it sufficient to just do:

    monster.missilestate = monster.seestate

?

File: examples/mood.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+The pistol has been replaced by the fist.
+All weapon pickups have been replaced by the Health Potion.
+
+For more information: https://jmtd.net/doom/mood/"""
+
+from deh9000 import *
+
+# pacifist monsters
+for monster in mobjinfo:
+    if not (monster.flags & MF_COUNTKILL or monster == mobjinfo[MT_SKULL]):
```
Comment (fragglet) [inline examples/mood.py]: Nit: 

    if not (monster.flags & MF_COUNTKILL) and monster != mobjinfo[MT_SKULL]:

**Review by fragglet — COMMENTED** [review]
Overall looks good. The rest of the codebase uses tabs for indents instead of spaces - do you want to adjust this to match?

**Comment by fragglet** [comment]
Re: pain elemental, let's just keep them pacifist as well. I suspect it's going to be hard to make them spawn lost souls without causing damage anyway.

File: examples/mood.py (hunk 1 above)
Comment (jmtd) [reply examples/mood.py]: sadly not, chocolate segfaults when the event occurs (I mean to separately file/fix that in chocolate)

Comment (jmtd) [reply examples/mood.py]: however there might be other, nicer approaches to achieve something similar. I was saving them for a future hack session.

**Comment by fragglet** [comment]
Thanks!

**Pull request merged**